org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandlerTest.java

```diff
@@ -3461,7 +3461,12 @@ public void testCompletion_Lambda() throws Exception {
 		assertNotNull(lambda);
 		assertTrue(lambda.getTextEdit().getLeft().getNewText().matches("\\$\\{1:\\w+\\} -> \\$\\{0\\}"));
 
-		assertEquals("(Object arg0) ->", lambda.getLabel());
+		try {
+			assertEquals("(Object t) ->", lambda.getLabel());
+		} catch (ComparisonFailure e) {
+			// In case the JDK has no sources
```

Comment:
If you can't beat 'em, join 'em :joy: . I seem to recall that resolving parameter names for a completion proposal can be expensive. Could it be that sometimes it just does't get resolved in time ? https://github.com/eclipse-jdt/eclipse.jdt.core/blob/7ddead3f30ea50adc762d55ec6a32568427e692c/org.eclipse.jdt.core/model/org/eclipse/jdt/core/CompletionProposal.java#L1622-L1624

Reply:
I have copied the following code - https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/31fb1b1d1751ce3e9548363672c5d570eea7eb75/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandlerTest.java#L723